status-reporter/status_reporter/rabbitmq_reporter.py

```diff
@@ -13,17 +13,110 @@
+from cloudify.cluster_status import (CloudifyNodeType, ServiceStatus,
+                                     NodeServiceStatus)
+
+from .status_reporter import Reporter, logger
+from .utils import get_systemd_services, get_node_status
+
+RABBITMQ_SERVICE_KEY = 'RabbitMQ'
+CA_PATH = '/etc/cloudify/ssl/rabbitmq-ca.pem'
+RABBITMQ_SERVICES = {'cloudify-rabbitmq.service': 'RabbitMQ'}
+RABBITMQ_URL = 'https://localhost:15671/api/healthchecks/node'
 
 
 class RabbitMQReporter(Reporter):
     def __init__(self):
         super(RabbitMQReporter, self).__init__(CloudifyNodeType.BROKER)
 
     def _collect_status(self):
-        return {}
+        services, statuses = get_systemd_services(RABBITMQ_SERVICES)
+        config = self._config('extra_config', None)
+        if self._rabbitmq_service_not_running(statuses) or not config:
+            return self._rabbitmq_collect_status_failed(services)
+        self._update_node_status(services, statuses, config)
+        self._update_cluster_status(services, statuses, config)
+        status = get_node_status(statuses)
+        return status, services
+
+    @staticmethod
+    def _rabbitmq_service_not_running(statuses):
+        return statuses[0] == NodeServiceStatus.INACTIVE
+
+    def _update_cluster_status(self, services, statuses, config):
+        cluster_status, cluster_extra_info = self._get_cluster_status(config)
+        services[RABBITMQ_SERVICE_KEY]['extra_info']['cluster_status'] = \
+            cluster_extra_info
+        statuses.append(cluster_status)
+
+    def _get_cluster_status(self, config):
+        cluster_status_info = self._rabbitmqctl_cluster_status(config)
+        if not cluster_status_info:
+            return NodeServiceStatus.INACTIVE, {}
+
+        return NodeServiceStatus.ACTIVE, cluster_status_info
+
+    def _rabbitmqctl_cluster_status(self, config):
+        cmd = self._get_cluster_status_cmd(config)
+        runner = LocalCommandRunner()
+        try:
+            cluster_status_info = json.loads(runner.run(cmd).std_out)
+        except CommandExecutionException as error:
+            logger.error(
+                'Failed getting RabbitMQ cluster-status due to '
+                '{0}'.format(error))
+            return None
+
+        return cluster_status_info
+
+    @staticmethod
+    def _get_cluster_status_cmd(config):
+        longnames = '--longnames' if config['use_long_name'] else ''
+        nodename = config['nodename']
+        return ('sudo rabbitmqctl -n {nodename} {longnames} cluster_status '
+                '--formatter json'.format(nodename=nodename,
+                                          longnames=longnames))
+
+    def _update_node_status(self, services, statuses, config):
+        node_status, node_extra_info = self._get_rabbitmq_node_status(config)
+        services[RABBITMQ_SERVICE_KEY]['extra_info']['node_status'] = \
+            node_extra_info
+        statuses.append(node_status)
+
+    def _get_rabbitmq_node_status(self, config):
```

Comment:
Like previous comment I think `_get_rabbitmq_node_status` and `_query_rabbitmq` can be merged to one function